src/Simulation/Common/QubitManager.cs

```diff
@@ -65,6 +69,7 @@ public QubitManager(long qubitCapacity, bool mayExtendCapacity = false, bool dis
             Debug.Assert(this.qubits[NumQubits - 1] == None);
 
             free = 0;
+            freeTail = NumQubits - 1;
```

Comment:
Given it's only valid for EncourageReuse=false, would it make sense to set it to -1 (or some other 'invalid' sentinel value) when EncourageReuse=true?

Reply:
I don't see why. It is only used in discourage reuse cases, but it is valid always. It is OK either way but it will take more churn to maintain -1 value. Every time you set it you have to do an if? Not worth it.

Reply:
The comment for freeTail was saying that it's only valid when EncourageReuse=false. When a state of a class is allowed to be inconsistent, I like making it more explicit with "invalid" markers so can assert on them and spot problems under debugger easier. In this case, probably should just change to comment to say "only _used_ when EncourageReuse=false".